### PR TITLE
Reimplement move_selected_chat_id_to_deltachat_controller and move

### PR DESCRIPTION
**Pull request by Jikstra**
archive chat functionality to backend

### CODE REVIEW

File: src/main/deltachat.js [hunk 1]
```diff
@@ -258,20 +260,18 @@ class DeltaChatController {
...
    * Dispatched from menu alternative in SplittedChatListAndView
    */
-  archiveChat (chatId) {
+  archiveChat (chatId, archive) {
     log('archiving chat', chatId)
-    this._dc.archiveChat(chatId, true)
+    this._dc.archiveChat(chatId, archive)
+  }
+
+
```
Comment (ralphtheninja) [inline src/main/deltachat.js]: Two lines

File: src/renderer/components/ChatList.js [hunk 2]
```diff
@@ -4,13 +4,14 @@ const { ConversationListItem } = require('./conversations')
 
 class ChatList extends React.Component {
   render () {
-    const { chats, selectedChatId } = this.props
+    const { chats, selectedChatId, showArchivedChats } = this.props
     const tx = window.translate
 
     if (!selectedChatId) {
+      let msg = showArchivedChats ? 'No archived chats' : 'You have no chats, feel free to start a new one :)'
```
Comment (ralphtheninja) [inline src/renderer/components/ChatList.js]: nice

File: src/renderer/components/SplittedChatListAndView.js [hunk 3]
```diff
@@ -46,64 +44,30 @@ class SplittedChatListAndView extends React.Component {
...
+    ipcRenderer.send('dispatch', 'deleteChat',  selectedChat.id)
   }
 
   onEditGroup (selectedChat) {
     this.props.changeScreen('EditGroup', { chatId: selectedChat.id, chatName: selectedChat.name })
   }
 
   onLeaveGroup (selectedChat) {
-    ipcRenderer.send('dispatch', 'leaveGroup', this.state.selectedChatId)
+    ipcRenderer.send('dispatch', 'leaveGroup',  selectedChat.id)
```
Comment (ralphtheninja) [inline src/renderer/components/SplittedChatListAndView.js]: Two spaces

File: src/renderer/components/ChatList.js (hunk 2 above)
Comment (karissa) [inline src/renderer/components/ChatList.js]: this needs to be internationalized

**Pull request merged**